File: solutions/java/bob/1/src/main/java/Bob.java
class Bob {
    String hey(String input) {
        StatementProperties properties = new StatementProperties(input);
        return properties.isSilent()
                       ? "Fine. Be that way!"
                       : properties.isForcefulQuestion()
                                 ? "Calm down, I know what I'm doing!"
                                 : properties.isShouting()
                                           ? "Whoa, chill out!"
                                           : properties.isQuestion()
                                                     ? "Sure."
                                                     : "Whatever.";
    }
}

File: solutions/kotlin/bob/1/src/main/kotlin/Bob.kt
object Bob {
	fun hey(input: String): String {
		val trimmedInput = input.trim()
		if (trimmedInput.isEmpty()) {
			return "Fine. Be that way!"
		}

		val isQuestion = trimmedInput.endsWith("?")
		val letters = trimmedInput.filter { it.isLetter() }
		val isYelling = letters.isNotEmpty()
		                && letters.all { it.isUpperCase() }

		return when {
			isQuestion && isYelling
				-> "Calm down, I know what I'm doing!"

			isQuestion
				-> "Sure."

			isYelling
				-> "Whoa, chill out!"

			else
				-> "Whatever."
		}
	}
}